src/data_extract.cc

```diff
@@ -0,0 +1,44 @@
+/*
+ * data_extract.cc
+ *
+ *  Created on: Jul 19, 2023
+ *      Author: rao
+ */
+#include<iostream>
+#include"data_extract.hpp"
+#include"data_header.hpp"
+
+
+//data_extract::data_extract():data_extract(0,0){}
+
+data_extract::data_extract()
+	{
+		header = 0xDEADBEEFBEEFFEED;
+	}
+void data_extract::set_header(unsigned long long int upd_header)
+	{
+		header = upd_header;
+	}
+
+
+std::string data_extract:: extract_data(int buf_siz , char *buff)
+	{
+		data_header dt1;
+		std::string the_msg;
+		memcpy(&dt1 , buff, sizeof(data_header));
```

Comment:
Now make an implementation, that uses zero copy here just to check the data

data_header *dt1;
dt1 =(data_header *) buff;
dt1->delimeter... and so on